smartsim/settings/__init__.py

```diff
@@ -32,69 +34,55 @@
 
 
 # TODO Mock imports for compiling tests
-class DragonRunSettings:
-    pass
+class SettingsBase:
+    def __init__(self, *_: t.Any, **__: t.Any) -> None: ...
```

Comment:
why did u define dunders for base?

Reply:
Just to get past the type check without ignoring all of `_core`. There are still a few places where settings are instanced using one of these stubclasses. If we do not define a constructor that takes any number of args and any number of kwargs, the type checker will complain anytime a settings is instanced that does not conform to the signature

Reply:
Similarly, the `__getattr__` is there so that the type checker does not complain when try to dot-access an attr that does not directly exist on these stub classes.